pkg/reconciler/delivery/delivery.go

```diff
@@ -99,53 +108,114 @@ func (c *Reconciler) updateRoute(ctx context.Context, cfg *v1.Configuration) err
+		if t.RevisionName == newRevName {
+			nameListed = true
+			break
+		}
+	}
+	if !nameListed {
+		return false
+	}
+	if len(route.Status.Traffic) == 1 || *route.Status.Traffic[1].Percent == 100 {
+		return true
+	}
+	for _, c := range route.Status.Conditions {
+		if c.Type != "IngressReady" {
+			continue
+		}
+		if isLastTransitionExpired(c.LastTransitionTime.Inner) {
+			return false
 		}
 	}
 	return false
 }
 
+// isLastTransitionExpired determines if enough time has elapsed since the last transition of the IngressReady condition
+func isLastTransitionExpired(ltt metav1.Time) bool {
+	// this is the tricky part, using time stamps to figure out if Route is up to date
+	// IngressReady transition has a short delay of a couple seconds, which demands some wiggle room (the -5)
+	// this sacrifices some timing precision, but I haven't thought of a better way yet
+	// it also poses problems when the user applies the 2nd Revision very soon (fewer than policy.Threshold
+	// seconds) after the 1st Revision
+	return time.Now().After(ltt.Add(time.Duration(policy.Threshold - 5) * time.Second))
+}
+
 // modifyRouteSpec is a toy function that is designed specifically for the proof-of-concept
 // it modifies the Route spec field to accommodate the new Revision, if necessary
 func modifyRouteSpec(route *v1.Route, newRevName string) (*v1.Route, error) {
 	// if there is currently zero traffic targets, then set the Configuration's
 	// latest ready Revision as the default traffic target
-	// if there is currently one traffic target, then split 50% off that target and
+	// if there is currently one traffic target, then split a certain % off that target and
 	// direct it to the new Revision
-	// if there are 2 or more traffic targets, return an error (unexpected use case)
-	if len(route.Status.Traffic) == 0 {
+	// if there are 2 traffic targets, update the percentage split, or report error if the 
+	// new Revision name doesn't match with either target
+	// Note: when there are > 1 traffic targets, it is assumed that they are ordered from oldest to newest
+	var newPercent int = 100
+	var err error
+
+	if len(route.Status.Traffic) == 1 {
+		if route.Status.Traffic[0].RevisionName == newRevName {
+			return route, nil
+		}
+		newPercent, err = computeNewPercent(&policy, 0)
+		if err != nil {
+			return route, err
+		}
+	} else if len(route.Status.Traffic) == 2 {
+		if route.Status.Traffic[0].RevisionName != newRevName && route.Status.Traffic[1].RevisionName != newRevName {
+			return nil, fmt.Errorf("Unsupported use case: current implementation only supports 2 Revisions at once")
+		}
+		newPercent, err = computeNewPercent(&policy, int(*route.Status.Traffic[1].Percent))
+		if err != nil {
+			return route, err
+		}
+	}
+
+	if newPercent == 100 {
 		route.Spec.Traffic = []v1.TrafficTarget{
 			{
 				ConfigurationName: route.Name, // assume namespace/name matches for Route & Config
 				LatestRevision: ptr.Bool(true),
 				Percent: ptr.Int64(100),
 			},
 		}
```

Comment:
You can `{{...}}` to save some whitespace. Like below.

Reply:
fixed